src/components/container/SwellingCalendar/SwellingCalendar.tsx

```diff
@@ -0,0 +1,177 @@
+import { FontAwesomeSvgIcon } from "react-fontawesome-svg-icon";
+export type SwellingCalendarPreviewState = "WithData" | "NoData" | "Loading" | "Live";
+
+export interface SwellingCalendarProps {
+    surveyName : string,
+    dateRecordedResultIdentifier : string, 
+    severityResultIdentifier : string, 
+    previewState? : SwellingCalendarPreviewState,
+}
+
+interface SwellingDay {
+    dateRecorded : string,
+    severity : string
+}
+
+export default function (props: SwellingCalendarProps) {
+    const surveyAnswerQuery : SurveyAnswersQuery = {
+        surveyName : props.surveyName,
+        resultIdentifier : [props.dateRecordedResultIdentifier, props.severityResultIdentifier]
+    }
+    const [datePagerMonth, setMonth] = useState<Date>(startOfMonth(new Date()));
+    const [data, setData] = useState<Map<string, SwellingDay> | undefined>();
+
+    async function initialize(){
+        if (!data){
+            if (props.previewState && props.previewState == "NoData")
+            {
+                transformToCalendarData([]);
+            } else {
+                if (props.previewState && props.previewState == "WithData"){
+                    transformToCalendarData(previewSwellingData);
+                }
+                else
+                {
+                    surveyResults(surveyAnswerQuery).then((results: SurveyAnswer[]) => {
+                        transformToCalendarData(results);
+                    });
+                }
+            }
+        }
+    }
+
+    const stateConfiguration: CalendarDayStateConfiguration = {
+        'mild': {
+            style: {
+                background: '#F7DA19'
+            },
+            streak: true,
+            streakColor: '#FDEF97'
+        },
+        'moderate': {
+            style: {
+                background: 'var(--mdhui-color-warning)'
+            },
+            streak: true,
+            streakColor: '#FCCCC7'
+        },
+        'severe': {
+            style: {
+                background: 'var(--mdhui-color-danger)'
+            },
+            streak: true,
+            streakColor: '#FCCCC7'
+        },
+    };
+
+    const computeStateForDay = (date: Date): string => {
+        let key : string = startOfDay(date).toDateString();
+        return (data && data?.get(key)?.severity) ?? "";
+    };
+
+    function transformToCalendarData(results : SurveyAnswer[]){
+        var swellingEntries : Map<Guid, SwellingDay> = createResultsMap(results);
+
+        var calendarData : Map<string, SwellingDay> = new Map<string, SwellingDay>();
+        swellingEntries.forEach( swellingEntry => {
+            let key : string = startOfDay(parseISO(swellingEntry.dateRecorded)).toDateString();
+            let exists = calendarData.get(key);
+            if (exists){
+                if (parseISO(swellingEntry.dateRecorded) > parseISO(exists.dateRecorded)){
+                    exists.severity = swellingEntry.severity;
+                    calendarData.set(key, exists);
```

Comment:
I'm not sure, but I don't think you need to set it again here.  The updated `exists` object will still be in the map with that key.